plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/explorer/OpenShiftExplorerContentProvider.java

```diff
@@ -65,7 +65,9 @@ protected void handleConnectionChanged(IConnection connection, String property,
 				cache.flushFor((Connection) newValue);
 			}
 			refreshViewer(newValue);
-		} 
+		} else if (ConnectionProperties.PROPERTY_RESOURCE.equals(property)) {
+		    refreshViewer(newValue);
```

Comment:
For me, it is the part of the modifications that fixed the problem. If you remove that, then if you reedit from the  Openshift explorer view after you modified the resource and closed the editor will bring you to the outdated resource.
The other part of the modifications are just there to make sure when use the base TextEditor in order to listen to both editor events and connection manager events